dali/kernels/erase/erase_cpu_test.cc

```diff
@@ -0,0 +1,165 @@
+      for (int i2 = 0; i2 < sh[2]; i2++) {
+        LOG_LINE << " " << data_view.data[k++];
+      }
+      if (sh[2] > 1) LOG_LINE << "\n";
+    }
+    LOG_LINE << "\n";
+  }
+  LOG_LINE << "\n";
+}
+
+template <typename T, int Dims>
+void VerifyErase(OutTensorCPU<T, Dims> out_view,
+                 OutTensorCPU<T, Dims> in_view,
+                 const TensorShape<Dims> &roi_anchor,
+                 const TensorShape<Dims> &roi_shape,
+                 const T* fill_values,
+                 int channels_dim = -1) {
+  ASSERT_EQ(out_view.shape, in_view.shape);
+  const auto &data_shape = out_view.shape;
+  for (int i0 = 0, k = 0; i0 < data_shape[0]; i0++) {
+    int roi_end_0 = (roi_anchor[0] + roi_shape[0]);
+    for (int i1 = 0; i1 < data_shape[1]; i1++) {
+      int roi_end_1 = (roi_anchor[1] + roi_shape[1]);
+      for (int i2 = 0; i2 < data_shape[2]; i2++, k++) {
+        int roi_end_2 = (roi_anchor[2] + roi_shape[2]);
+        bool erased =
+             i0 >= roi_anchor[0] && i0 < roi_end_0
+          && i1 >= roi_anchor[1] && i1 < roi_end_1
+          && i2 >= roi_anchor[2] && i2 < roi_end_2;
+        int c = -1;
+        if (channels_dim == 0) {
+          c = i0;
+        } else if (channels_dim == 1) {
+          c = i1;
+        } else if (channels_dim == 2) {
+          c = i2;
+        }
+        auto fill_value = channels_dim < 0 ? fill_values[0] : fill_values[c];
+        auto expected_value = erased ? fill_value : in_view.data[k];
+        EXPECT_EQ(expected_value, out_view.data[k])
+          << make_string(i0, ",", i1, ",", i2, " roi_start: ",
+                         roi_anchor[0], ",", roi_anchor[1], ",", roi_anchor[2], " roi_end ",
+                         roi_end_0, ",", roi_end_1, ",", roi_end_2,
+                         " erased ", erased);
+      }
+    }
+  }
+}
+TEST_P(EraseCpuTest, EraseCpuTest) {
+  using T = float;
+  constexpr int Dims = 3;
+
+  auto shape = in_view_.shape;
+  auto size = volume(shape);
+
+  KernelContext ctx;
+  kernels::EraseArgs<T, Dims> args;
+  args.rois = {{roi_anchor_, roi_shape_}};
+  if (channels_dim_ > 0) {
+    args.rois[0].channels_dim = channels_dim_;
+    args.rois[0].fill_values.clear();
+    for (auto v : fill_values_) {
+      args.rois[0].fill_values.push_back(v);
+    }
+  }
+
+  kernels::EraseCpu<T, Dims> kernel;
+  auto req = kernel.Setup(ctx, in_view_, args);
+
+  // Shape should be the same as the input
+  ASSERT_EQ(shape, req.output_shapes[0][0]);
+  std::vector<T> out(size, 0.0f);
+  auto out_view = OutTensorCPU<T, Dims>(out.data(), shape.to_static<Dims>());
+
+  kernel.Run(ctx, out_view, in_view_, args);
+
+  LOG_LINE << "in:\n";
+  print_data(in_view_);
+
+  LOG_LINE << "out:\n";
+  print_data(out_view);
+
+  VerifyErase(out_view, in_view_, roi_anchor_, roi_shape_,
+              fill_values_.data(), channels_dim_);
+}
+
+INSTANTIATE_TEST_SUITE_P(EraseCpuTest, EraseCpuTest, testing::Combine(
+    testing::Values(std::array<int64_t, 3>{32, 15, 3}),  // data shape
```

Comment:
Can you add a test case of 1C image as well?

Reply:
Will do